var/spack/repos/builtin/packages/glib/package.py

```diff
@@ -85,6 +102,32 @@ def fix_python_path(self):
                     '#!/usr/bin/env python',
                     *files)
 
+    @run_before('configure')
+    def fix_dtrace_usr_bin_path(self):
+        if 'tracing=dtrace' not in self.spec:
+            return
+
+        # dtrace may cause glib build to fail because it uses
+        # '/usr/bin/python' in the shebang. To work around that
+        # we copy the original script into a temporary folder, and
+        # change the shebang to '/usr/bin/env python'
+        dtrace = which('dtrace').path
```

Comment:
I think the whole hack could be applied iff the `dtrace` command is found *and* it actually contains the fixed sbang but this would lead to a silently ignored `tracing=dtrace` variant. Does Spack have a policy for this kind of situations?

Reply:
> Does Spack have a policy for this kind of situations (e.g.: never ever ignore a variant, just let it fail)?

Not really, at least as far as I know. @adamjstewart do you have an opinion on this one?

In any case, the *should be applied if the command is found* part is kind of already there - if `dtrace` is not found then `which` returns `None` and asking for `.path` will make it fail (not in a nice way and not with a nice message though).

Reply:
No opinions.